### ISSUE TITLE
New UXG Not showing data

### ISSUE THREAD

**Issue by lslamp** [unpoller/unpoller#678]
I have replaced my USG for the new UXG lite. The swap out went well and my unifi network is working without any issues.
The dashboard was working before the swap out but stopped after.
Looking in the logs it concerns me that it shows Gateway = 0

Below are the rows that concern me.
I assume that this following row shows that the UXG device is seen.
2024-02-29T21:17:26.146616+01:00 ICANUNIFI unpoller[586624]: 2024/02/29 21:17:26.146510 updateweb.go:222: [DEBUG] Unmarshalling Device Type: uxg, Model: UXG, site ICAN (default)

In the following row it seems that the Gateways show 0. This does not seem correct.
2024-02-29T21:17:26.241519+01:00 ICANUNIFI unpoller[586624]: 2024/02/29 21:17:26.240855 logger.go:19: [INFO] UniFi Metrics Recorded. Site: 1, Client: 39, **Gateways: 0**, UAP: 3, USW: 7, IDS/Event/Alarm/Anomaly: 0/0/0/0, DPI Site/Client: 1/62, Point: 1180, Fields: 7123, Err: 0, Dur: 82ms

Do you have any idea how to get this fixed?
Thanks
Lawrence

Below is the full extract. 
2024-02-29T21:16:56.239573+01:00 ICANUNIFI unpoller[586624]: 2024/02/29 21:16:56.239284 logger.go:19: [INFO] UniFi Metrics Recorded. Site: 1, Client: 39, Gateways: 0, UAP: 3, USW: 7, IDS/Event/Alarm/Anomaly: 0/0/0/0, DPI Site/Client: 1/62, Point: 1180, Fields: 7123, Err: 0, Dur: 86ms
2024-02-29T21:17:26.049104+01:00 ICANUNIFI unpoller[586624]: 2024/02/29 21:17:26.048386 updateweb.go:222: [DEBUG] Collecting controller data: https://127.0.0.1:8443 ()
2024-02-29T21:17:26.049431+01:00 ICANUNIFI unpoller[586624]: 2024/02/29 21:17:26.048446 updateweb.go:222: [DEBUG] Polling controller: https://127.0.0.1:8443 ()
2024-02-29T21:17:26.049631+01:00 ICANUNIFI unpoller[586624]: 2024/02/29 21:17:26.048573 updateweb.go:222: [DEBUG] Requesting https://127.0.0.1:8443/api/stat/sites, with params: false, cookies: 2
2024-02-29T21:17:26.053840+01:00 ICANUNIFI unpoller[586624]: 2024/02/29 21:17:26.053741 updateweb.go:222: [DEBUG] Requested https://127.0.0.1:8443/api/stat/sites: elapsed 5ms, returned 1964 bytes
2024-02-29T21:17:26.054059+01:00 ICANUNIFI unpoller[586624]: 2024/02/29 21:17:26.054005 updateweb.go:222: [DEBUG] Found 1 site(s): default
2024-02-29T21:17:26.054248+01:00 ICANUNIFI unpoller[586624]: 2024/02/29 21:17:26.054035 updateweb.go:222: [DEBUG] Polling Controller, retreiving Site DPI data, site ICAN (default)
2024-02-29T21:17:26.054416+01:00 ICANUNIFI unpoller[586624]: 2024/02/29 21:17:26.054075 updateweb.go:222: [DEBUG] Requesting https://127.0.0.1:8443/api/s/default/stat/sitedpi, with params: true, cookies: 2
2024-02-29T21:17:26.069260+01:00 ICANUNIFI unpoller[586624]: 2024/02/29 21:17:26.069163 updateweb.go:222: [DEBUG] Requested https://127.0.0.1:8443/api/s/default/stat/sitedpi: elapsed 15ms, returned 14661 bytes
2024-02-29T21:17:26.070573+01:00 ICANUNIFI unpoller[586624]: 2024/02/29 21:17:26.070422 updateweb.go:222: [DEBUG] Polling Controller, retreiving Client DPI data, site ICAN (default)
2024-02-29T21:17:26.070705+01:00 ICANUNIFI unpoller[586624]: 2024/02/29 21:17:26.070527 updateweb.go:222: [DEBUG] Requesting https://127.0.0.1:8443/api/s/default/stat/stadpi, with params: true, cookies: 2
2024-02-29T21:17:26.095446+01:00 ICANUNIFI unpoller[586624]: 2024/02/29 21:17:26.095303 updateweb.go:222: [DEBUG] Requested https://127.0.0.1:8443/api/s/default/stat/stadpi: elapsed 25ms, returned 231481 bytes
2024-02-29T21:17:26.105410+01:00 ICANUNIFI unpoller[586624]: 2024/02/29 21:17:26.105216 updateweb.go:222: [DEBUG] Polling Controller, retreiving UniFi Clients, site ICAN (default)
2024-02-29T21:17:26.105586+01:00 ICANUNIFI unpoller[586624]: 2024/02/29 21:17:26.105511 updateweb.go:222: [DEBUG] Requesting https://127.0.0.1:8443/api/s/default/stat/sta, with params: false, cookies: 2
2024-02-29T21:17:26.110007+01:00 ICANUNIFI unpoller[586624]: 2024/02/29 21:17:26.109912 updateweb.go:222: [DEBUG] Requested https://127.0.0.1:8443/api/s/default/stat/sta: elapsed 4ms, returned 56152 bytes
2024-02-29T21:17:26.112046+01:00 ICANUNIFI unpoller[586624]: 2024/02/29 21:17:26.111940 updateweb.go:222: [DEBUG] Requesting https://127.0.0.1:8443/api/s/default/stat/device, with params: false, cookies: 2
2024-02-29T21:17:26.138928+01:00 ICANUNIFI unpoller[586624]: 2024/02/29 21:17:26.136932 updateweb.go:222: [DEBUG] Requested https://127.0.0.1:8443/api/s/default/stat/device: elapsed 25ms, returned 236821 bytes
2024-02-29T21:17:26.140296+01:00 ICANUNIFI unpoller[586624]: 2024/02/29 21:17:26.140188 updateweb.go:222: [DEBUG] Unmarshalling Device Type: usw, Model: US8, site ICAN (default)
2024-02-29T21:17:26.141352+01:00 ICANUNIFI unpoller[586624]: 2024/02/29 21:17:26.141246 updateweb.go:222: [DEBUG] Unmarshalling Device Type: uap, Model: U7MP, site ICAN (default)
2024-02-29T21:17:26.142486+01:00 ICANUNIFI unpoller[586624]: 2024/02/29 21:17:26.142408 updateweb.go:222: [DEBUG] Unmarshalling Device Type: usw, Model: US8, site ICAN (default)
2024-02-29T21:17:26.143179+01:00 ICANUNIFI unpoller[586624]: 2024/02/29 21:17:26.143090 updateweb.go:222: [DEBUG] Unmarshalling Device Type: usw, Model: US8, site ICAN (default)
2024-02-29T21:17:26.144001+01:00 ICANUNIFI unpoller[586624]: 2024/02/29 21:17:26.143881 updateweb.go:222: [DEBUG] Unmarshalling Device Type: usw, Model: US8P60, site ICAN (default)
2024-02-29T21:17:26.144981+01:00 ICANUNIFI unpoller[586624]: 2024/02/29 21:17:26.144857 updateweb.go:222: [DEBUG] Unmarshalling Device Type: uap, Model: UAP6MP, site ICAN (default)
2024-02-29T21:17:26.146616+01:00 ICANUNIFI unpoller[586624]: 2024/02/29 21:17:26.146510 updateweb.go:222: [DEBUG] Unmarshalling Device Type: uxg, Model: UXG, site ICAN (default)
2024-02-29T21:17:26.151444+01:00 ICANUNIFI unpoller[586624]: 2024/02/29 21:17:26.151324 updateweb.go:222: [DEBUG] Unmarshalling Device Type: usw, Model: US8P60, site ICAN (default)
2024-02-29T21:17:26.152346+01:00 ICANUNIFI unpoller[586624]: 2024/02/29 21:17:26.152236 updateweb.go:222: [DEBUG] Unmarshalling Device Type: usw, Model: US8P60, site ICAN (default)
2024-02-29T21:17:26.153268+01:00 ICANUNIFI unpoller[586624]: 2024/02/29 21:17:26.153161 updateweb.go:222: [DEBUG] Unmarshalling Device Type: usw, Model: US8, site ICAN (default)
2024-02-29T21:17:26.154248+01:00 ICANUNIFI unpoller[586624]: 2024/02/29 21:17:26.154064 updateweb.go:222: [DEBUG] Unmarshalling Device Type: uap, Model: U7PG2, site ICAN (default)
2024-02-29T21:17:26.155454+01:00 ICANUNIFI unpoller[586624]: 2024/02/29 21:17:26.155343 updateweb.go:222: [DEBUG] Collecting controller events: https://127.0.0.1:8443 ()
2024-02-29T21:17:26.155567+01:00 ICANUNIFI unpoller[586624]: 2024/02/29 21:17:26.155410 updateweb.go:222: [DEBUG] Requesting https://127.0.0.1:8443/api/stat/sites, with params: false, cookies: 2
2024-02-29T21:17:26.159165+01:00 ICANUNIFI unpoller[586624]: 2024/02/29 21:17:26.159026 updateweb.go:222: [DEBUG] Requested https://127.0.0.1:8443/api/stat/sites: elapsed 4ms, returned 1964 bytes
2024-02-29T21:17:26.159312+01:00 ICANUNIFI unpoller[586624]: 2024/02/29 21:17:26.159259 updateweb.go:222: [DEBUG] Found 1 site(s): default
2024-02-29T21:17:26.241519+01:00 ICANUNIFI unpoller[586624]: 2024/02/29 21:17:26.240855 logger.go:19: [INFO] UniFi Metrics Recorded. Site: 1, Client: 39, Gateways: 0, UAP: 3, USW: 7, IDS/Event/Alarm/Anomaly: 0/0/0/0, DPI Site/Client: 1/62, Point: 1180, Fields: 7123, Err: 0, Dur: 82ms

**Comment by lslamp** [comment]
@platinummonkey Sorry if this is a stupid question mate, but what does added the bug label actually mean?

**Comment by platinummonkey** [comment]
Ah I'm just classifying issues, I've spent a little time looking at this one but it's not clear to me where the bug is yet

**Comment by lslamp** [comment]
@platinummonkey Not sure if this is relevant, but when I was looking around in the code area in the file listing. I did see that there was a file called uxg.go ... now my installation is pretty old, so it might be that I am running an older version of unpoller. IT might not be compiled with the uxg libraries.
How do I upgrade my installation to being it to the latest if it is at all an older version.

my version is UniFi Poller v2.9.5 Starting Up! PID: 1091076

Lawrence

**Comment by platinummonkey** [comment]
2.9.5 is a recent release (2.10.0 was just released so you didn't miss anything there). 

How about the Unifi software versions: UnifiOS & Network version - to see if these are recent as well.

**Comment by lslamp** [comment]
I am currently using the new interface 
This is what I see.
UniFi Network Application 8.0.28
![image](https://github.com/unpoller/unpoller/assets/6013759/a43342dd-febe-49b6-9491-d2dee3466fc6)
Lawrence

**Comment by lslamp** [comment]
@platinummonkey Hey fella, sorry to hassle you but have you figured anything out with this issue?

Do you have any ideas?

Lawrence

**Comment by platinummonkey** [comment]
I hadn’t yet - you might do a unpoller -d run maybe there is something special in the uxg output section

**Comment by lslamp** [comment]
@platinummonkey 
what do you mean by unpoller -d. Can I simply run this on the commandline.
Not sure if it is relevant but I went through all the unifi logs and I see absolutely no reference to UXG in any way or form. I have opened a ticket with unifi on how or what to look for within the unifi logs to understand what it is referencing with talking to teh device.

Annoying that they deliver a device with no SNMP and no log entries.
Lawrence

**Comment by lslamp** [comment]
AH sorry I am new to this.

I ran the -d option and this is the output.

root@ICANUNIFI:~# unpoller -d
2024/03/19 12:21:30 [INFO] Loading Configuration File: /etc/unpoller/up.conf
2024/03/19 12:21:30 [INFO] Checking inputs...
2024/03/19 12:21:30 [INFO]      (1/1) Checking input unifi...
2024/03/19 12:21:30 [INFO]               unifi is OK
2024/03/19 12:21:30 [INFO] Checking outputs...
2024/03/19 12:21:30 [INFO]      (1/5) Checking output WebServer...
2024/03/19 12:21:30 [INFO]               WebServer is OK
2024/03/19 12:21:30 [INFO]      (2/5) Checking output datadog...
2024/03/19 12:21:30 [INFO]               datadog is OK
2024/03/19 12:21:30 [INFO]      (3/5) Checking output influxdb...
2024/03/19 12:21:30 [INFO]               influxdb is OK
2024/03/19 12:21:30 [INFO]      (4/5) Checking output loki...
2024/03/19 12:21:30 [INFO]               loki is OK
2024/03/19 12:21:30 [INFO]      (5/5) Checking output prometheus...
2024/03/19 12:21:30 [INFO]               prometheus is OK
2024/03/19 12:21:30 Failed debug checks


Lawrence

**Comment by platinummonkey** [comment]
Ah sorry “-j” this dumps the json from the device response (it’s long)

**Comment by lslamp** [comment]
@platinummonkey 
No clue what I did but I ssh to the gateway and ran apt update and then apt upgrade, there were a number of debian packages that needed to be upgraded. After the upgrade I forced rebooted the device and then checked the dashboard and bingo. I had the gateway visible. No clue how or why.
![image](https://github.com/unpoller/unpoller/assets/6013759/99105370-19f5-4b27-8a13-ccbb7de8b6d3)

I do have a concern/question.
It seems that the Angular is depreciated and will stop being used in future versions of grafana. How can this be fixed?

This warning is being shown on almost all of teh dashboard panels.
Thanks
Lawrence
